index.d.ts

```diff
@@ -0,0 +1,249 @@
+declare module 'ember-concurrency' {
+  import Task from 'ember-concurrency/task';
+
+  type TaskDecorator = PropertyDecorator & {
+    drop(): TaskDecorator;
+    keepLatest(): TaskDecorator;
+    restartable(): TaskDecorator;
+    maxConcurrency(num: number): TaskDecorator;
+    enqueue(): TaskDecorator;
+    withTestWaiter(): PropertyDecorator;
+  };
+
+  export function task(
+    generator: Function
+  ): PropertyDecorator & {
+    drop(): TaskDecorator;
+    keepLatest(): TaskDecorator;
+    restartable(): TaskDecorator;
+    maxConcurrency(num: number): TaskDecorator;
+    enqueue(): TaskDecorator;
+    withTestWaiter(): PropertyDecorator;
+  };
+  export function timeout(wait: number): Promise<void>;
+
+  export function didCancel(error: Error): boolean;
```

Comment:
This should probably be `export function didCancel(reason: unknown): reason is Error;`. In a promise callback, I don't think you necessarily know the reject reason object is an `Error`, and so if you require this parameter to be `Error` TypeScript might not let you pass it.